ui/src/app/services/sync.notifier.ts

```diff
@@ -71,12 +81,81 @@ export class SyncNotifier {
         version: server.versionInstalled,
       },
     })
-    this.osWelcomeOpen = true
 
-    modal.onWillDismiss().then(() => {
-      this.osWelcomeOpen = false
-      return this.apiService.acknowledgeOSWelcome(this.config.version)
+    modal.onDidDismiss().then(() => {
+      this.apiService.acknowledgeOSWelcome(this.config.version)
+      this.handleUpdateCheck(server)
     })
     await modal.present()
   }
+
+  private async handleUpdateCheck (server: Readonly<S9Server>) {
+    if (!server.autoCheckUpdates || this.checkedForUpdates) return
+
+    this.checkedForUpdates = true
+
+    if (server.versionLatest && this.emver.compare(server.versionInstalled, server.versionLatest) === -1) {
+      return this.presentAlertNewOS(server.versionLatest)
```

Comment:
if this errors checkedForUpdates is still true

Reply:
if someone decides not to upgrade the OS, will they ever see the alert for new apps should their be some?